dexlib2/src/main/java/org/jf/dexlib2/builder/LocatedItems.java

```diff
@@ -0,0 +1,89 @@
+package org.jf.dexlib2.builder;
+
+import com.google.common.collect.ImmutableList;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import java.util.*;
+
+public abstract class LocatedItems<T extends ItemWithLocation> {
+    @Nullable
+    private List<T> items = null;
+
+    @Nonnull
+    private List<T> getMutableItems() {
+        if (items == null) {
+            items = new ArrayList<>(1);
+        }
+        return items;
+    }
+
+    @Nonnull
+    private List<T> getImmutableItems() {
```

Comment:
"getImmutableItems" implies that it returns an ImmutableList, which isn't always true.

The simplest thing would be to get rid of the whole mutable vs. immutable, but iirc, I did this because it was too costly to create an empty list for every instruction, especially when most of them would remain empty anyway.

Maybe just rename this to "getItems", since the returned list may or may not be mutable. Which of course means it should only be used when the returned list doesn't need to be changed.

Reply:
Agreed.
It was confusing, so I also changed and renamed `getMutableItems` into `initItemsIfNull`.